src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts

```diff
@@ -161,4 +180,23 @@ export class CreateProviderComponent extends CreateFormComponent implements OnIn
       }
     });
   }
+
+  logout(){
+    if (!this.user.isRegistered) {
+    
+        const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+          width: Constants.MODAL_SMALL,
+          data: {
+            type: ModalConfirmationType.leaveRegistration,
+            property: '',
+          },
+        });
+    
```

Comment:
I think you can remove the subscription earlier and use the approach to take a snapshot
     const isRegistered = this.store.selectSnapshot(RegistrationState.user).isRegistered;
and then use this value for checking